streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessorSupplier.java

```diff
@@ -150,16 +153,16 @@ public void init(final org.apache.kafka.streams.processor.ProcessorContext conte
         }
 
         @Override
-        public void process(final K key, final Change<V> value) {
+        public void process(final Record<K, Change<V>> record) {
             observedStreamTime = Math.max(observedStreamTime, internalProcessorContext.timestamp());
-            buffer(key, value);
+            buffer(record);
             enforceConstraints();
         }
 
-        private void buffer(final K key, final Change<V> value) {
-            final long bufferTime = bufferTimeDefinition.time(internalProcessorContext, key);
+        private void buffer(final Record<K, Change<V>> record) {
+            final long bufferTime = bufferTimeDefinition.time(internalProcessorContext, record.key());
 
-            buffer.put(bufferTime, key, value, internalProcessorContext.recordContext());
+            buffer.put(bufferTime, record.key(), record.value(), internalProcessorContext.recordContext());
```

Comment:
key and value, sure. For record context, I'm not really sure how to replace it. Its usage its propagated down to `BufferValue` and not only record metadata, but timestamp, and --most importantly--ser/des are used at the bottom.
Replacing it will involve refactoring all the buffering.

Reply:
Thanks, @jeqo , I can see that my tone in that comment didn't encompass the scope of the change I was proposing.

I'm fine if we want to delay that refactor for later work and focus on the Processor API itself in this PR.